projects/or_scs/src/Mirza/OrgRegistry/Types.hs

```diff
@@ -291,29 +291,58 @@ data LocationResponse = LocationResponse
   , geoLocAddress :: Maybe Text
   } deriving (Show, Generic, Eq)
 instance ToSchema LocationResponse
-instance ToJSON LocationResponse
-instance FromJSON LocationResponse
+instance ToJSON LocationResponse where
+  toJSON (LocationResponse locId locGLN pfix geolocId geolocCoord geolocAddr) = object
+    [ "location_id" .= locId
+    , "epc" .= locGLN
+    , "company_prefix" .= pfix
+    , "geolocation_id" .= geolocId
+    , "geolocation_coords" .= geolocCoord
+    , "geolocation_address" .= geolocAddr
+    ]
 
+instance FromJSON LocationResponse where
+  parseJSON = withObject "LocationResponse" $ \o -> LocationResponse
+    <$> o .: "location_id"
+    <*> o .: "epc"
+    <*> o .: "company_prefix"
+    <*> o .: "geolocation_id"
+    <*> o .:? "geolocation_coords"
+    <*> o .: "geolocation_address"
 
 data OrgAndLocationResponse = OrgAndLocationResponse
   { orgResponse      :: OrgResponse
   , locationResponse :: LocationResponse
   } deriving (Show, Generic, Eq)
 instance ToSchema OrgAndLocationResponse
-instance ToJSON OrgAndLocationResponse
-instance FromJSON OrgAndLocationResponse
-
+instance ToJSON OrgAndLocationResponse where
+  toJSON (OrgAndLocationResponse org loc) = object
+    [ "org" .= org
+    , "loc" .= loc
+    ]
+instance FromJSON OrgAndLocationResponse where
+  parseJSON = withObject "OrgAndLocationResponse" $ \o -> OrgAndLocationResponse
+    <$> o .: "org"
+    <*> o .: "location"
```

Comment:
These don't appear to be complimentary: "loc" vs "location". For reference in the public interface I strongly believe that why shouldn't be abbreviating as I think this aids with readability (even if there are other considerations like bandwitdh etc). Another thing to consider is to used "named constants" (or similar technique) to make sure that the JSON key strings are the same in toJSON and fromJSON, to prevent this type of issue happening. This comment applies everywhere. e.g.
```
orgAndLocationResponseOrgKey = "org"
orgAndLocationResponseLocKey = "loc"

instance ToJSON OrgAndLocationResponse where
  toJSON (OrgAndLocationResponse org loc) = object
    [ orgAndLocationResponseOrgKey .= org
    , orgAndLocationResponseLocKey .= loc
    ]
instance FromJSON OrgAndLocationResponse where
  parseJSON = withObject "OrgAndLocationResponse" $ \o -> OrgAndLocationResponse
    <$> o .: orgAndLocationResponseOrgKey
    <*> o .: orgAndLocationResponseLocKey
```